TheCraigHewitt/Seriously-Simple-Podcasting
# Feature/series graphic

## Summary

This PR can be summarized in the following changelog entry:

* Adds a protected text domain unique identifier attribute to the admin class. Changes uses of the string found in the admin class to use the new protected attribute.
* Adds a metabox to the Series term add/edit forms for attaching an image to a series.
* Enqueues the WordPress media library editor only for the Series term add/edit forms.
* Adds a series image thumbnail to the series term list.

## Relevant technical choices:

* Changed the text domain strings to using a protected class attribute to ensure consistent use in future development.
* A default series display only image has been included as a placeholder only, presented when no series image has been assigned.
* No changes were made to podcast feeds or Series listings to present the series image at this time. Further uses, requiring additional code changes, need to be confirmed with the plugin maintainers.

## Test instructions

This PR can be tested by following these steps:

* New series:
  1. Add a new Series.
  2. Click new "Add Image" button.
  3. Upload and/or choose an image.
  4. Save the series and confirm editing and saving still work as expected.

* Edit an existing Series:
  1. Choose "Edit" on an existing Series entry.
  2. Upload or change the Series image.
  3. Update the chosen Series and confirm the series graphic is presenting as expected.

Fixes #139
